packages/jest-circus/src/eventHandler.ts

```diff
@@ -112,14 +107,14 @@ const eventHandler: EventHandler = (event, state): void => {
 
       if (type === 'beforeAll') {
         invariant(describeBlock, 'always present for `*All` hooks');
-        addErrorToEachTestUnderDescribe(describeBlock, error, asyncError);
+        addErrorToEachTestUnderDescribe(describeBlock!, error, asyncError);
       } else if (type === 'afterAll') {
         // Attaching `afterAll` errors to each test makes execution flow
         // too complicated, so we'll consider them to be global.
         state.unhandledErrors.push([error, asyncError]);
       } else {
         invariant(test, 'always present for `*Each` hooks');
-        test.errors.push([error, asyncError]);
+        test!.errors.push([error, asyncError]);
```

Comment:
Is it possible to type `invariant` so that TS understands it'll throw?

Reply:
As I know TS does not support throw type.

Reply:
Yeah, you're right. https://github.com/Microsoft/TypeScript/issues/10421